### PR TITLE
Updating doc links to centre on puppet-lint to work on githubpages

### PR DESCRIPTION
**Pull request by davidsandilands**
Currently all the links in github pages are broken, I am hoping this update should make the links functional

### CODE REVIEW

**Review by ekohl — CHANGES_REQUESTED** [review]
Jekyll has a concept of `baseurl` which I think is better.

https://mademistakes.com/mastering-jekyll/site-url-baseurl/ has more, but the short summary is:

* Set `baseurl: "/puppet-lint"` in `_config.yml`
* Change all links to `{{ site.baseurl }}/PATH`

Now if you want to move back to the top level, you simply change the config once. It also works with `jekyll serve`.

**Comment by ekohl** [comment]
See https://github.com/puppetlabs/puppet-lint/pull/77 for what I meant. As you can see, with `bundle exec jekyll serve` it will serve as `http://127.0.0.1:4000/puppet-lint/` and have the correct stylesheet and links (where I used baseurl at least).

File: _layouts/default.html [hunk 1]
```diff
@@ -41,10 +41,10 @@
             <a href="/plugins/">Community Plugins</a>
           </li>
           <li>
-            <a href="https://github.com/rodjek/puppet-lint/blob/master/CHANGELOG.md">Changelog</a>
+            <a href="https://github.com/puppetlabs/puppet-lint/blob/master/CHANGELOG.md">Changelog</a>
```
Comment (ekohl) [inline _layouts/default.html:44]:
```suggestion
            <a href="https://github.com/puppetlabs/puppet-lint/blob/main/CHANGELOG.md">Changelog</a>
```

**Comment by davidsandilands** [comment]
@ekohl thanks for that suggestion that is a much better idea. I will rework the PR

File: developer/tutorial/index.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ at some options. When you find one you're happy with (the MIT license is highly
 Gem::Specification.new do |spec|
   spec.name        = 'puppet-lint-trailing_newline-check'
   spec.version     = '1.0.0'
-  spec.homepage    = 'https://github.com/rodjek/puppet-lint-trailing_newline-check'
+  spec.homepage    = 'https://github.com/puppetlabs/puppet-lint-trailing_newline-check'
```
Comment (ekohl) [inline developer/tutorial/index.md:57]: In this case it refers to an existing external repo (and many of the same links in this file). I'd not update them.

**Comment by davidsandilands** [comment]
Thanks @ekohl for better approach on this. I have updated and forked so the github pages can be previewed/tested here https://davidsandilands.github.io/puppet-lint/

**Review by ekohl — APPROVED** [review]
I didn't review it in detail, but linkchecker is happy and a quick read through doesn't show anything odd.
```
$ linkchecker https://davidsandilands.github.io/puppet-lint/
INFO linkcheck.cmdline 2023-02-15 13:14:28,576 MainThread Checking intern URLs only; use --check-extern to check extern URLs.
LinkChecker 10.1.0
Copyright (C) 2000-2016 Bastian Kleineidam, 2010-2021 LinkChecker Authors
LinkChecker comes with ABSOLUTELY NO WARRANTY!
This is free software, and you are welcome to redistribute it under
certain conditions. Look at the file `LICENSE' within this distribution.
Read the documentation at https://linkchecker.github.io/linkchecker/
Write comments and bugs to https://github.com/linkchecker/linkchecker/issues

Start checking at 2023-02-15 13:14:28+002
10 threads active,     2 links queued,    6 links in  18 URLs checked, runtime 1 seconds
10 threads active,    24 links queued,   55 links in  89 URLs checked, runtime 6 seconds
10 threads active,     7 links queued,   72 links in  89 URLs checked, runtime 11 seconds
 1 thread active,     0 links queued,   92 links in  93 URLs checked, runtime 16 seconds

Statistics:
Downloaded: 349.23KB.
Content types: 1 image, 45 text, 0 video, 0 audio, 1 application, 0 mail and 46 other.
URL lengths: min=17, max=112, avg=67.

That's it. 93 links in 93 URLs checked. 0 warnings found. 0 errors found.
Stopped checking at 2023-02-15 13:14:45+002 (17 seconds)
```

**Pull request merged**